CDCgov/prime-reportstream
# Platform/mkalish/8669

This PR adds a new CLI command `migratekeys` to support moving keys from senders to the organization

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. First make sure that you have senders with API keys stored with them (can verify with `./gradlew primeCLI --args="sender get --name='{sender_name}'"`
2. Run `./gradlew primeCLI --args="organization migratekeys --orgName='{orgName}' "` and verify the output
3. Run `./gradlew primeCLI --args="organization migratekeys --orgName='{orgName}' --doit"`
4. Run `./gradlew primeCLI --args="sender get --name='{sender_name}'"` and verify the sender no longer has associated keys

## Changes
- Adds a new temporary `migratekeys` command to move keys from senders to their organization

## Checklist

### Testing

### Process

## Linked Issues
- Fixes https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/8669

## To Be Done
- Use this command to move keys in staging and then production after verification

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
